mypyc/test-data/run-classes.test

```diff
@@ -4088,3 +4088,435 @@ def test_inheritance_2() -> None:
     x = None
     y = None
     d = None
+
+[case testDunderGetAttr]
+from mypy_extensions import mypyc_attr
+from typing import Dict, Optional
+
+class GetAttr:
+    class_var = "x"
+
+    def __init__(self, extra_attrs: Dict[str, object], regular_attr: int):
```

Comment:
Style nit: no need to use `Optional` or `Dict`, since `t | None` and `dict[...]` work on all supported Python versions.